### PR TITLE
ETCD-666: Add etcd-backup-server pruning logic

### PR DESCRIPTION
**Pull request by Elbehery**
resolves https://issues.redhat.com/browse/ETCD-666

### CODE REVIEW

File: pkg/cmd/backuprestore/backupserver.go [hunk 1]
```diff
@@ -42,11 +48,16 @@ type backupServer struct {
...
+	prune.PruneOpts
 }
 
 func NewBackupServer(ctx context.Context) *cobra.Command {
 	backupSrv := &backupServer{
 		backupOptions: backupOptions{errOut: os.Stderr},
+		PruneOpts: prune.PruneOpts{
+			RetentionType: "None",
+			BackupPath:    backupVolume,
+		},
```
Comment (tjungblu) [inline pkg/cmd/backuprestore/backupserver.go:60]: do we need to set this?

File: pkg/cmd/prune-backups/prune.go [hunk 2]
```diff
@@ -33,10 +33,14 @@ type PruneOpts struct {
 	RetentionType      string
 	MaxNumberOfBackups int
 	MaxSizeOfBackupsGb int
+	BackupPath         string
```
Comment (tjungblu) [inline pkg/cmd/prune-backups/prune.go:36]: can you change the comment at the top to reflect that we pass a path now?

File: pkg/cmd/prune-backups/prune.go [hunk 3]
```diff
@@ -33,10 +33,14 @@ type PruneOpts struct {
...
 }
 
 func NewPruneCommand() *cobra.Command {
-	opts := PruneOpts{RetentionType: "None"}
+	opts := PruneOpts{
+		RetentionType: "None",
+		BackupPath:    BasePath,
```
Comment (tjungblu) [inline pkg/cmd/prune-backups/prune.go:42]: let's make this a cmdline arg of the prune opts, you can still default to BasePath

File: pkg/cmd/prune-backups/prune_test.go [hunk 4]
```diff
@@ -54,15 +54,15 @@ func TestRetainByNumber(t *testing.T) {
 		time.Sleep(1 * time.Second)
 	}
 
-	require.NoError(t, retainByNumber(6))
```
Comment (tjungblu) [inline pkg/cmd/prune-backups/prune_test.go:57]: would you mind adding some tests that will test varying folders instead of hardcoding BasePath?

Comment (tjungblu) [reply pkg/cmd/prune-backups/prune_test.go:57]: what also works for me: we do not have to change BasePath anymore, we can simply inject the tmp folder into the arguments

File: pkg/cmd/prune-backups/prune.go (hunk 3 above)
Comment (tjungblu) [reply pkg/cmd/prune-backups/prune.go:42]: and make `BasePath` a constant then, I think we can get away without the unit tests changing the path as a variable

File: pkg/cmd/backuprestore/backupserver.go (hunk 1 above)
Comment (Elbehery) [reply pkg/cmd/backuprestore/backupserver.go:60]: so the backup path is const [here](https://github.com/openshift/cluster-etcd-operator/blob/735f37b8d3938a03cdf1c74e61f634a8adcf97c2/pkg/cmd/prune-backups/prune.go#L18)

In order to user Prune, and use the NoConfig backup [volume](https://github.com/openshift/cluster-etcd-operator/blob/735f37b8d3938a03cdf1c74e61f634a8adcf97c2/pkg/cmd/backuprestore/backupserver.go#L18), I override the path here

please correct me if I am wrong

File: pkg/cmd/prune-backups/prune.go (hunk 3 above)
Comment (Elbehery) [reply pkg/cmd/prune-backups/prune.go]: done 👍🏽

File: pkg/cmd/backuprestore/backupserver.go (hunk 1 above)
Comment (tjungblu) [reply pkg/cmd/backuprestore/backupserver.go:60]: but is this correct?
because you're setting the backupDir path to "backupVolume + dateString"

Comment (tjungblu) [reply pkg/cmd/backuprestore/backupserver.go:60]: nvm, that's just the suffix for each backup invocation

Comment (Elbehery) [reply pkg/cmd/backuprestore/backupserver.go:60]: genauuu :D :D

**Comment by tjungblu** [comment]
/override ci/prow/e2e-aws-ovn-etcd-scaling

unrelated version operator failure

**Pull request merged**